### PR TITLE
yazi: fix HEAD build

### LINKED ISSUE #415: issue installing head version of yazi on macos
### What system are you running Yazi on?

macOS

### What terminal are you running Yazi in?

kitty 0.31.0

### Yazi version

yazi 0.1.5

### Did you try the latest main branch to see if the problem has already been fixed?

Tried, but the problem is still present

### Describe the bug

i'm encountering difficulties installing the head version of yazi on macos venture 13.6.1. i had successfully installed the head version about a month ago, but the current attempt is yielding an error related to cargo, stating that it's referring to a non-existent directory.

steps taken:
uninstalled the previous head version.
attempted to reinstall, but the issue persists.

error message:
```sh
==> cargo install --path=app
Last 15 lines from /Users/bassam/Library/Logs/Homebrew/yazi/01.cargo:
2023-12-01 03:53:58 +0000

cargo
install
--locked
--root=/usr/local/Cellar/yazi/HEAD-d549831
--path=app

error: `/private/tmp/yazi-20231130-3501-gu5u6c/app` is not a directory. --path must point to a directory containing a Cargo.toml file.

```

additional notes:
i understand that head versions can be unstable, and i apologize for opening this issue. the features i see are compelling, and i would like to explore them at my own pace. this is not a high-priority matter.

thank you for your understanding.

### Expected Behavior

installing HEAD successfully

### To Reproduce

`brew install yazi --HEAD`

### Configuration

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sxyazi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Fixes https://github.com/sxyazi/yazi/issues/415

### CODE REVIEW

**Comment by sxyazi** [comment]
Ah I didn't realize that the stable version is also built via `install` instead of using pre-built binaries directly.

In versions before v0.1.5, it was the `app` directory, but it was changed to `yazi-fm` to satisfy the publishing requirements on `crates.io`. Is there anyone give me some tips on how to handle it?

File: Formula/y/yazi.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Yazi < Formula
   depends_on "rust" => :build
 
   def install
-    system "cargo", "install", *std_cargo_args(path: "app")
+    system "cargo", "install", *std_cargo_args(path: "yazi-fm")
```
Comment (SMillerDev) [inline Formula/y/yazi.rb:22]:
```suggestion
    path = if build.head?
      "yazi-fm"
    else
      "app"
    end
    system "cargo", "install", *std_cargo_args(path: path)
```

Comment (sxyazi) [reply Formula/y/yazi.rb]: Thank you!

**Comment by chenrui333** [comment]
> > are those pre-release tags added recently?
> 
> @chenrui333 Nope, it has been there all along, not added recently.
> 
> This was a mistake from v0.1.0 - I mistakenly checked "Set as a pre-release" and this mistake has persisted since then. I will remove that tag in the next version (v0.1.6), please add it to the allowed list for now.

sounds good, let me refresh your PR with the allowlist change.

**Comment by chenrui333** [comment]
actually, since this is head build fix, we can just ignore the audit failure for now.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
just audit check failure, https://github.com/Homebrew/homebrew-core/actions/runs/7063745977/job/19230493212?pr=156077

rerun with syntax only label

**Review by p-linnane — APPROVED** [review]

**Pull request merged**